modules/core/src/main/java/org/apache/synapse/mediators/elementary/Target.java

```diff
@@ -421,10 +474,15 @@ private void setEnrichResultToBody(MessageContext synapseContext, SynapseJsonPat
         String newJsonString = "";
 
         if (action.equalsIgnoreCase(ACTION_REPLACE)) {
-            // replaces an existing value in json
-            newJsonString = JsonPath.parse(jsonString).set(expression, sourceNode).jsonString();
+            if (isRootPath) {
+                log.error("Acting replace is not supported for root path in type custom. " +
```

Comment:
Shall we check the possibility to move this validation to deployment

Reply:
Done